internal/pkg/meta/internal/adv/talos/talos.go

```diff
@@ -95,7 +95,7 @@ func (a *ADV) Unmarshal(buf []byte) error {
 
 	magic2 := binary.BigEndian.Uint32(buf[len(buf)-4:])
 	if magic2 != Magic2 {
-		return fmt.Errorf("adv: unexpected magic %x, expecting %x", magic2, Magic2)
+		return fmt.Errorf("adv: unexpected magic %x, expecting %x", magic2, uint32(Magic2))
```

Comment:
Don't know why but this started failing on build with:
```
 > [talosctl-linux-armv7-build 2/3] RUN --mount=type=cache,target=/.cache GOOS=linux GOARCH=arm GOARM=7 go build  -ldflags "-s -w" -o /talosctl-linux-armv7:
#119 0.646 # github.com/siderolabs/talos/internal/pkg/meta/internal/adv/talos
#119 0.646 /src/internal/pkg/meta/internal/adv/talos/talos.go:98:71: cannot use Magic2 (untyped int constant 2780087250) as int value in argument to fmt.Errorf (overflows)
```

Reply:
`GOARCH=arm` is 32bit arm. 2_780_087_250 > math.Int (2^31) there, so this makes sense.

Reply:
I tried it locally, it fails for `GOARCH=386` too.

Reply:
Smallest fix in #6980